src/ripple/rpc/impl/RPCHandler.cpp

```diff
@@ -173,6 +174,30 @@ Status callMethod (
             jtGENERIC, "cmd:" + name);
 
         auto ret = method (context, result);
+        // Handle system-level warnings on admin connections
+        if (context.role == Role::ADMIN)
```

Comment:
I don't know that we want this _here_, touching _every_ command. I think it should be in the `server_info` handler.

Reply:
Yeah, I have concerns about putting the warning in every command.

Reply:
My thinking. rippled currently has the following indications that a server is amendment blocked (https://xrpl.org/amendments.html#amendment-blocked)
1. When submitting a transaction, it returns `amendmentBlocked`.
2. It logs a `LedgerMaster:ERR` message.
3. `server_info` returns a flag.

I figure that being amendment blocked is pretty drastic, and the user is likely to notice pretty quickly, so those notifications are sufficient. But the warning is less drastic, because everything will continue to work normally until it "suddenly" doesn't, so I wanted the warnings to be more aggressive so the user has more of a chance to notice them.  Thus, logging the `LedgerMaster:ERR` message every flag ledger, and including `warnings` on every admin-level RPC command instead of just `submit` and `server_info`. I'm not married to the idea, though, so if you still think it's not a good idea to have it on every comment, I'll move the check, but I would like to still maintain at minimum returning the `warnings` on both `submit` and `server_info`.

Reply:
@nbougalis Per our conversation, the warning is now only included in the results for `server_info` and `server_state`, and added a second warning for amendment blocked with a more verbose description of the problem.